aem-authoring-toolkit-plugin/src/main/java/com/exadel/aem/toolkit/plugin/util/ClassUtil.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.exadel.aem.toolkit.plugin.util;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Member;
+import java.lang.reflect.Modifier;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import com.exadel.aem.toolkit.api.annotations.widgets.accessory.Ignore;
+import com.exadel.aem.toolkit.api.annotations.widgets.accessory.IgnoreFields;
+import com.exadel.aem.toolkit.api.handlers.Source;
+import com.exadel.aem.toolkit.plugin.adapters.ClassMemberSetting;
+import com.exadel.aem.toolkit.plugin.source.Sources;
+import com.exadel.aem.toolkit.plugin.util.ordering.OrderingUtil;
+
+/**
+ * Contains utility methods for parsing AEM components' classes and extracting information related to UI rendering
+ */
+public class ClassUtil {
+
+    /**
+     * Default (hiding) constructor
+     */
+    private ClassUtil() {
+    }
+
+    /**
+     * Retrieves a sequential list of {@link Source} objects representing manageable members that belong
+     * a certain {@code Class} (and its superclasses) and match specific criteria
```

Comment:
`match specific criteria` - perhaps we can clarify , which criteria?

Reply:
Fixed